src/js/common/schemaform/FormApp.jsx

```diff
@@ -41,19 +47,35 @@ class FormApp extends React.Component {
       window.History.scrollRestoration = 'manual';
     }
 
-    // If we start in the middle of a form, redirect to the beginning
+    // If we start in the middle of a form, redirect to the beginning or load
+    //  saved form / prefill
     const currentPath = this.props.currentLocation.pathname;
     const firstPagePath = this.props.routes[this.props.routes.length - 1].pageList[0].path;
     // If we're in production, we'll redirect if we start in the middle of a form
-    // In development, we won't redirect unless we append the URL with `?redirect` ()
+    // In development, we won't redirect unless we append the URL with `?redirect`
     const devRedirect = __BUILDTYPE__ !== 'development' || this.props.currentLocation.search.includes('redirect');
     if (currentPath !== firstPagePath && devRedirect) {
-      // If the first page is not the intro and uses `depends`, this will probably break
-      this.props.router.push(firstPagePath);
+      // We started on a page that isn't the first, so after we know whether
+      //  we're logged in or not, we'll load or redirect as needed.
+      this.shouldRedirectOrLoad = true;
+    }
+  }
+
+  componentDidMount() {
+    // When a user isn't logged in, the profile finishes loading before the component mounts
+    if (!this.props.profileIsLoading && this.shouldRedirectOrLoad) {
+      this.redirectOrLoad(this.props);
     }
   }
 
   componentWillReceiveProps(newProps) {
+    // When a user is logged in, the profile finishes loading after the component
+    //  has mounted, so we check here.
+    // If we're done loading the profile, check to see if we should load or redirect
+    if (this.props.profileIsLoading && !newProps.profileIsLoading && this.shouldRedirectOrLoad) {
```

Comment:
Should we reset `shouldRedirectOrLoad` once this condition is hit? It seems like this might get tripped inadvertently if the session expires and the user logs in again.